ports/rp2/machine_uart.c

```diff
@@ -305,7 +295,7 @@ STATIC mp_obj_t machine_uart_make_new(const mp_obj_type_t *type, size_t n_args,
     }
 
     // Initialise the UART peripheral if any arguments given, or it was not initialised previously.
-    if (n_args > 1 || n_kw > 0 || self->baudrate == 0) {
+    if (n_args > 0 || self->baudrate == 0) {
```

Comment:
I think this needs `kw_args->used > 0` added

Reply:
Fixed.